src/components/pages/LockScreen.tsx

```diff
@@ -53,10 +53,46 @@ export const LockScreen = (): JSX.Element => {
   const biometryAvailable = Boolean(flagshipMetadata.biometry_available)
   const biometryType = flagshipMetadata.biometry_type
 
-  const onBiometryLock = (): void =>
-    void handleChange(setBiometry, 'biometryLock', webviewIntent).then(
-      value => value && setAutoLock(true)
-    )
+  const onBiometryLock = (): void => {
+    if (!webviewIntent) return
+
+    Promise.all([
+      !biometryEnabled &&
+        webviewIntent.call(
```

Comment:
nit: I think I'll add a guideline to reject `false && foo()` syntax. if/else clarity for the win.

Reply:
it's still handy when you can't easily use an if statement